Automattic/sensei
# Add svg icons sprite

Fixes #4482

### Changes proposed in this Pull Request

* This PR has the goal to refactor the use of icons in different places, and a big cleanup.
* It's a mix of 2 approaches we were using:
  * It introduces a [webpack plugin](https://github.com/cascornelissen/svg-spritemap-webpack-plugin) to generate a sprite from an icons library (folder `assets/images/svg-icons`).
  * So we continue using sprites, but in an external file, making it reusable and cacheable.
  * And we continue with a function (`Sensei()->assets->get_icon( 'icon_name', 'css-classes to-the-svg' )`) to get the SVG HTML to use in the PHP, that gives us the code to use a specific icon in the sprite.
* More context on alternatives we thought: p1639751782020800-slack-C02KZBEJ30D
* It also introduces the [`@svgr/webpack`](https://react-svgr.com/docs/webpack/) dependency to use the same SVG files in the React side (it transpiles svg files to components), so we don't need to duplicate them to use in React.
* It also removes some `@wordpress/icons` duplication, that existed in order to support WordPress 5.3.

### Testing instructions


* Build the project, and make sure the SVG icons are working properly in the frontend, and in the editor.


### Deprecated Code

* `Sensei_Course_Outline_Course_Block::render_svg_icon_library`
